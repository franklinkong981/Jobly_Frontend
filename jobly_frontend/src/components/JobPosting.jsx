import React from "react";

function JobPosting() {
  return (

  );
}

export default JobPosting;