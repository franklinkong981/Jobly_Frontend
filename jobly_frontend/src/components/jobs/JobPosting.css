.JobPosting {
  border: 2px solid red;
  background-color: lightblue;
}