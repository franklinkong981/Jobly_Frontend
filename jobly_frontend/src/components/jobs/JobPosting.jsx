import React from "react";
import { Card, CardBody, CardTitle} from "reactstrap";

function JobPosting({jobName}) {

  return (
    <section className="col-md-8">
      <Card>
        <CardBody className="text-center">
          <CardTitle>
            {jobName}
          </CardTitle>
        </CardBody>
      </Card>
    </section>
  );
}

export default JobPosting;