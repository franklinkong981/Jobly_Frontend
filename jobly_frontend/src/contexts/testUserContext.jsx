import React from "react";
import CurrentUserContext from "./currentUserContext.jsx";

const testUser = {
  username: "testuser",
  first_name: "Testy",
  last_name: "Usery",
  email: "testuser981@gmail.com"
};

const TestUserProvider = ({children, currentTestUser = testUser, hasUserAppliedToJob= () => false}) => (
  <CurrentUserContext.Provider value={{currentTestUser, hasUserAppliedToJob}}>
    {children}
  </CurrentUserContext.Provider>
);

export {TestUserProvider};