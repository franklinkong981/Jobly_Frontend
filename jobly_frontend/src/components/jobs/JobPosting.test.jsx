import React from "react";
import {render} from "@testing-library/react";

import {TestUserProvider} from "../../contexts/testUserContext.jsx";

import JobPosting from "./JobPosting.jsx";
import { expect, it } from "vitest";

const testJob = {title: "Tester", companyName: "VS Code", salary: 100000, equity: 10};

it("renders without crashing", function() {
  render(
    <TestUserProvider>
      <JobPosting job={testJob} />
    </TestUserProvider>
  );
});

it("matches snapshot", function() {
  const {asFragment} = render(
    <TestUserProvider>
      <JobPosting job={testJob} />
    </TestUserProvider>
  );
  expect(asFragment()).toMatchSnapshot();
});