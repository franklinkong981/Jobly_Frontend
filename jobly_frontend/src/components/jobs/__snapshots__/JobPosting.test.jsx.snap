// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="JobPosting card"
  >
    <div
      class="JobPosting-content card-body"
    >
      <h3
        class="JobPosting-title card-title"
      >
        Tester
      </h3>
      <h6
        class="JobPosting-company"
      >
        VS Code
      </h6>
      <p>
        Salary: $100,000
      </p>
      <p>
        Equity: 10
      </p>
      <button
        class="btn btn-danger font-weight-bold float-right"
      >
        Apply
      </button>
    </div>
  </div>
</DocumentFragment>
`;
